pkg/build/bazel/bazel.go

```diff
@@ -18,16 +18,43 @@ import (
 	"os"
 	"path"
 	"path/filepath"
+	"strconv"
 	"strings"
 
 	inner "github.com/bazelbuild/rules_go/go/tools/bazel"
+	inner_testutil "github.com/bazelbuild/rules_go/go/tools/bzltestutil"
 )
 
+// bazelTestEnvVar can be used to determine when running in the `bazel test`
+// environment.
+const bazelTestEnvVar = "BAZEL_TEST"
+
 // Return true iff this library was built with Bazel.
 func BuiltWithBazel() bool {
 	return true
 }
 
+// Return true iff called from a test run by Bazel.
+func TestByBazel() bool {
+	if bazelTestEnv, ok := os.LookupEnv(bazelTestEnvVar); ok {
+		if bazelTest, err := strconv.ParseBool(bazelTestEnv); err == nil {
+			return bazelTest
+		}
+	}
+
+	return false
+}
+
+// Return true iff called from a test using Bazel's Go test wrapping.
```

Comment:
I don't like this function name and the comment is pretty confusing. A reason for this is that there isn't a wrapper and a test binary: they are the same binary, just run with different environment variables.